### PR TITLE
feat: optimize Dockerfile in cross-modal-search and fixes #685

### LINKED ISSUE #685: cross-modal-search: 'NumpyIndexer' object has no attribute '_embedding_matrix'
A member from the community is having issue running the cross-modal-search example.

The error that he encountered is as follows:

image

## Affected Examples
cross-modal-search

### PR DESCRIPTION
**Pull request by winstonww**
This PR optimizes the Dockerfile in cross-modal-search example and closes issue #685. Thanks.

### CODE REVIEW

File: cross-modal-search/Dockerfile [hunk 1]
```diff
@@ -2,12 +2,16 @@ FROM jinaai/jina:2.0.8-perf
 
 RUN apt-get -y update && apt-get install -y git wget gcc unzip libmagic1
 
-COPY . /workspace
 WORKDIR /workspace
+COPY ./requirements.txt requirements.txt
+
+RUN pip3 install torch==1.9.0+cpu torchvision==0.10.0+cpu torchaudio==0.9.0 -f https://download.pytorch.org/whl/torch_stable.html
```
Comment (numb3r3) [inline cross-modal-search/Dockerfile:8]: do we need to install `torchaudio==0.9.0`?

**Review by numb3r3 — CHANGES_REQUESTED** [review]

File: cross-modal-search/Dockerfile (hunk 1 above)
Comment (winstonww) [reply cross-modal-search/Dockerfile]: Nope, neither is torch vision required. I have removed them both. Please see [2978104](https://github.com/jina-ai/examples/pull/684#:~:text=fix%3A%20address%20comments-,2978014,-Merge%20state) for the change.

File: cross-modal-search/pods/executors.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 from PIL import Image
 import io
 from jina import Executor, DocumentArray, requests, Document
-from jina.helloworld.chatbot.my_executors import _norm, _ext_B, _ext_A, _cosine
+from jina.helloworld.multimodal.my_executors import _norm, _ext_B, _ext_A, _cosine
```
Comment (numb3r3) [inline cross-modal-search/pods/executors.py:11]: copy the code of `_norm, _ext_B, _ext_A, _cosine`

File: cross-modal-search/Dockerfile [hunk 3]
```diff
@@ -2,12 +2,16 @@ FROM jinaai/jina:2.0.8-perf
...
 RUN apt-get -y update && apt-get install -y git wget gcc unzip libmagic1
 
-COPY . /workspace
 WORKDIR /workspace
+COPY ./requirements.txt requirements.txt
+
+RUN pip3 install torch==1.9.0+cpu -f https://download.pytorch.org/whl/torch_stable.html
 
 # install the third-party requirements
 RUN pip install git+https://github.com/openai/CLIP.git click && pip uninstall -y jina && pip install -r requirements.txt && pip uninstall -y dataclasses && python -c "import clip; clip.load('ViT-B/32')"
```
Comment (numb3r3) [inline cross-modal-search/Dockerfile:11]: remove `pip uninstall -y jina`

**Review by numb3r3 — CHANGES_REQUESTED** [review]

File: cross-modal-search/pods/executors.py (hunk 2 above)
Comment (winstonww) [reply cross-modal-search/pods/executors.py]: Addressed, please see [here](https://github.com/jina-ai/examples/pull/684/commits/b941d959c228d88758aff43dfcf2a7004965f2db) for the change.

File: cross-modal-search/Dockerfile (hunk 3 above)
Comment (winstonww) [reply cross-modal-search/Dockerfile]: Addressed, please see [here](https://github.com/jina-ai/examples/pull/684/commits/b941d959c228d88758aff43dfcf2a7004965f2db) for the change, thank you.

**Review by numb3r3 — APPROVED** [review]
LGTM

**Pull request merged**